File: MorkosiaPrepaLeague/Models/Core/Team.cs
using System.ComponentModel.DataAnnotations;

namespace MorkosiaPrepaLeague.Models.Core
{
    public class Team : CosmosDocumentBase
    {
        [Required]
        public string LeagueId { get; set; } = string.Empty;

        [Required]
        public string ManagerUserId { get; set; } = string.Empty;

        [Required]
        public string TransferWindowId { get; set; } = string.Empty;

        public List<PlayerTeamPosition> Players { get; set; } = new List<PlayerTeamPosition>();

        /// <summary>
        /// Generates a deterministic ID based on manager and transfer window to ensure one team per manager per window
        /// </summary>
        /// <param name="managerUserId">The manager's user ID</param>
        /// <param name="transferWindowId">The transfer window ID</param>
        /// <returns>A composite ID for the team</returns>
        public static string GenerateTeamId(string managerUserId, string transferWindowId)
        {
            return $"{managerUserId}_{transferWindowId}";
        }

        /// <summary>
        /// Sets the team ID using the composite key approach
        /// </summary>
        public void SetCompositeId()
        {
            if (!string.IsNullOrEmpty(ManagerUserId) && !string.IsNullOrEmpty(TransferWindowId))
            {
                Id = GenerateTeamId(ManagerUserId, TransferWindowId);
            }
        }
    }

    public class PlayerTeamPosition
    {
        public string PlayerId { get; set; } = string.Empty;
        public string Position { get; set; } = string.Empty;
    }
}

File: src/app/models/team.py
from typing import List

from beanie import Indexed
from pydantic import BaseModel

from app.models.base import DocumentBase


class PlayerTeamPosition(BaseModel):
    # A single player slot inside a team.

    player_id: str = ""
    position: str = ""


class Team(DocumentBase):
    # The `id` is a composite string: "{manager_user_id}_{transfer_window_id}".
    # This enforces one team per manager per transfer window, exactly as in .NET.

    league_id: Indexed(str) = ""  # type: ignore[valid-type]
    manager_user_id: str = ""
    transfer_window_id: str = ""
    players: List[PlayerTeamPosition] = []

    class Settings:
        name = "teams"

    # ------------------------------------------------------------------
    # Composite ID helpers (ported from Team.cs)
    # ------------------------------------------------------------------

    @staticmethod
    def generate_team_id(manager_user_id: str, transfer_window_id: str) -> str:
        return f"{manager_user_id}_{transfer_window_id}"

    def set_composite_id(self) -> None:
        if self.manager_user_id and self.transfer_window_id:
            self.id = self.generate_team_id(self.manager_user_id, self.transfer_window_id)
